docs/reference/adaptors.rst

```diff
@@ -383,6 +383,39 @@ Adaptors
         * :func:`flux::scan`
         * :func:`flux::fold_first`
 
+``set_union``
+^^^^^^^^^^^^^
+
+..  function::
+    template <sequence Seq1, sequence Seq2, typename Cmp = std::ranges::less> \
+        requires see_below
+    auto set_union(Seq1&& seq1, Seq2&& seq2, Cmp cmp = {}) -> sequence auto;
+
+    Returns a sequence adaptor which yields the set union of the two input sequences :var:`seq1` and :var:`seq2`, ordered by the given comparison function :var:`cmp`.
+
+    This function assumes that both :var:`seq1` and :var:`seq2` are sorted in ascending order with respect to the comparison function :var:`cmp`. If the input sequences are not sorted, the contents of the resulting sequence is unspecified.
+
+    When the resulting sequence is iterated, it will output the elements from the two input sequences in order according to :var:`cmp`. If some element is found `m` times in :var:`seq1` and `n` times in :var:`seq2`, then the resulting sequence yields exactly `max(n, m)` elements.
+
+    :param seq1: The first sequence to merge.
+    :param seq2: The second sequence to merge.
+    :param cmp: A binary predicate that takes two elements as arguments and returns true if the first element is less than the second.
+    :requires: Both :var:`Seq1` and :var:`Seq2` must have common l-value and r-value reference, common value type and must be equality comparable with respect to `Cmp`.
```

Comment:
Please could you move the `:requires:` section above `:param seq1:`.

I think we need to come up with some terminology for "compatible sequences" which we can use in the documentation, because this is going to be used in several places (not just the set adaptors but `chain` as well). Leave it with me. :)

Reply:
Moved.